### DISCUSSION TITLE
Issue with stt in my deepgram-ai-agent-demo code

### DISCUSSION THREAD

**Discussion by satish-code-gif** [deepgram/community#861] [General help]
### Which Deepgram product feature are you using?

Deepgram API - TTS

### Details

this is route.ts code  return await fetch(
    `${process.env.DEEPGRAM_STT_DOMAIN}/v1/speak?model=${model}`,
    {
      method: "POST",
      body: JSON.stringify({ text }),
      headers: {
        "Content-Type": `application/json`,
        Authorization: `token ${process.env.DEEPGRAM_API_KEY || ""}`,
        "X-DG-Referrer": url,
      },
    }
  )
  
  and this is conversation.tsx code { const requestTtsAudio = useCallback(
    async (message: Message) => {
      const start = Date.now();
      const model = ttsOptions?.model ?? "aura-asteria-en";

      const res = await fetch(`/api/speak?model=${model}`, {
        cache: "no-store",
        method: "POST",
        body: JSON.stringify(message),
      });

      const headers = res.headers;

      const blob = await res.blob();

      startAudio(blob, "audio/mp3", message.id).then(() => {
        addAudio({
          id: message.id,
          blob,
          latency: Number(headers.get("X-DG-Latency")) ?? Date.now() - start,
          networkLatency: Date.now() - start,
          model,
        });
      });
    },} 
    
    here everything is working fine but stt is not working 

### If you are making a request to the Deepgram API, what is the full Deepgram URL you are making a request to?

i'm hitting with tsx code

### If you are making a request to the Deepgram API and have a request ID, please paste it below:

i'm hitting with tsx code

### If possible, please attach your code or paste it into the text box.

this is route.ts code  return await fetch(
    `${process.env.DEEPGRAM_STT_DOMAIN}/v1/speak?model=${model}`,
    {
      method: "POST",
      body: JSON.stringify({ text }),
      headers: {
        "Content-Type": `application/json`,
        Authorization: `token ${process.env.DEEPGRAM_API_KEY || ""}`,
        "X-DG-Referrer": url,
      },
    }
  )
  
  and this is conversation.tsx code { const requestTtsAudio = useCallback(
    async (message: Message) => {
      const start = Date.now();
      const model = ttsOptions?.model ?? "aura-asteria-en";

      const res = await fetch(`/api/speak?model=${model}`, {
        cache: "no-store",
        method: "POST",
        body: JSON.stringify(message),
      });

      const headers = res.headers;

      const blob = await res.blob();

      startAudio(blob, "audio/mp3", message.id).then(() => {
        addAudio({
          id: message.id,
          blob,
          latency: Number(headers.get("X-DG-Latency")) ?? Date.now() - start,
          networkLatency: Date.now() - start,
          model,
        });
      });
    },} 
    
    here everything is working fine but stt is not working

**Comment by team-deepgram** [comment]
Thanks for asking your question about Deepgram! If you didn't already include it in your post, please be sure to add as much detail as possible so we can assist you efficiently, such as:
- The `request_id` if you have a question about your requests or transcription responses.
- The features you used or the full `api.deepgram.com` URL you sent your request to, including parameters.
- Any code snippets you can share.

**Reply by satish-code-gif** [reply]
I already code snippet check once and that is my typescript code

**Comment by SandraRodgers** [comment]
Can you clarify what it is you are asking? I'm seeing text-to-speech being used but you say that it is STT (speech-to-text) that isn't working?

Also can you provide the error response you see when you run the code?

**Comment by satish-code-gif** [comment]
i'm asking about text-to-speech only.
